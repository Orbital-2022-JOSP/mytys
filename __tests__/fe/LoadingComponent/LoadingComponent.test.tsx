import { render } from '@testing-library/react';
import LoadingComponent from '../../../components/LoadingComponent/LoadingComponent';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Loading Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<LoadingComponent />);

            expect(container).toMatchSnapshot()
        })
})

