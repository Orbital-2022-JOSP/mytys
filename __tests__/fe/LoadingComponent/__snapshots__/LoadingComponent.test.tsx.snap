// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loading Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <div
      class="max-w-xl sm:mx-auto lg:max-w-2xl"
    >
      <div
        class="flex flex-col mb-16 sm:text-center sm:mb-0"
      >
        <div
          class="flex items-center justify-center w-120 h-60 rounded-full"
        >
          <div>
            <svg
              height="400"
              preserveAspectRatio="xMidYMid meet"
              style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
              viewBox="0 0 400 400"
              width="400"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <clippath
                  id="__lottie_element_2"
                >
                  <rect
                    height="400"
                    width="400"
                    x="0"
                    y="0"
                  />
                </clippath>
                <clippath
                  id="__lottie_element_7"
                >
                  <path
                    d="M0,0 L100,0 L100,150 L0,150z"
                  />
                </clippath>
                <clippath
                  id="__lottie_element_38"
                >
                  <path
                    d="M0,0 L400,0 L400,400 L0,400z"
                  />
                </clippath>
                <clippath
                  id="__lottie_element_57"
                >
                  <path
                    d="M0,0 L400,0 L400,400 L0,400z"
                  />
                </clippath>
              </defs>
              <g
                clip-path="url(#__lottie_element_2)"
              >
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,200)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M0.05000000074505806,-83.75 C46.33100128173828,-83.75 83.8499984741211,-46.23099899291992 83.8499984741211,0.05000000074505806 C83.8499984741211,46.33100128173828 46.33100128173828,83.8499984741211 0.05000000074505806,83.8499984741211 C-46.23099899291992,83.8499984741211 -83.75,46.33100128173828 -83.75,0.05000000074505806 C-83.75,-46.23099899291992 -46.23099899291992,-83.75 0.05000000074505806,-83.75z"
                      fill="rgb(0,176,255)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M2.8889999389648438,49 C2.8889999389648438,49 -4.888999938964844,49 -4.888999938964844,49 C-9.583000183105469,49 -13.388999938964844,45.194000244140625 -13.388999938964844,40.5 C-13.388999938964844,40.5 -13.388999938964844,40.5 -13.388999938964844,40.5 C-13.388999938964844,35.805999755859375 -9.583000183105469,32 -4.888999938964844,32 C-4.888999938964844,32 2.8889999389648438,32 2.8889999389648438,32 C7.583000183105469,32 11.388999938964844,35.805999755859375 11.388999938964844,40.5 C11.388999938964844,40.5 11.388999938964844,40.5 11.388999938964844,40.5 C11.388999938964844,45.194000244140625 7.583000183105469,49 2.8889999389648438,49z"
                      fill="rgb(255,79,0)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M59,-26 C59,-26 59.5,43.67300033569336 59.5,43.67300033569336 C53.582000732421875,42.45399856567383 43.95000076293945,40.891998291015625 32,41 C16.917999267578125,41.137001037597656 13.805999755859375,43.79199981689453 -1.2519999742507935,43.672000885009766 C-15.446999549865723,43.558998107910156 -17.80900001525879,41.159000396728516 -32,41 C-43.895999908447266,40.867000579833984 -53.512001037597656,42.428001403808594 -59.5,43.67300033569336 C-59.5,20.448999404907227 -59.5,-2.7760000228881836 -59.5,-26"
                      fill-opacity="0"
                      stroke="rgb(255,79,0)"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-opacity="1"
                      stroke-width="8"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M56.76100158691406,41 C51.108001708984375,40.090999603271484 42.112998962402344,38.95800018310547 31,39 C15.57699966430664,39.05799865722656 12.996000289916992,41.31399917602539 -1,41 C-15.788000106811523,40.667999267578125 -18.18400001525879,38.03099822998047 -32,38 C-42.941001892089844,37.97600173950195 -51.882999420166016,39.60499954223633 -57.76100158691406,41 C-57.76100158691406,41 -57.76100158691406,-27.357999801635742 -57.76100158691406,-27.357999801635742 C-57.76100158691406,-27.357999801635742 56.76100158691406,-27.357999801635742 56.76100158691406,-27.357999801635742 C56.76100158691406,-27.357999801635742 56.76100158691406,41 56.76100158691406,41z"
                      fill="rgb(227,242,252)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M-58,-27 C-58,-27 -47,-35 -47,-35 C-47,-35 -47,31 -47,31 C-47,31 -58,41 -58,41 C-58,41 -58,-27 -58,-27z"
                      fill="rgb(237,246,252)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M45.91699981689453,-35.01599884033203 C45.91699981689453,-35.01599884033203 56.33700180053711,-28.28700065612793 56.33700180053711,-28.28700065612793 C56.33700180053711,-28.28700065612793 56.33700180053711,39 56.33700180053711,39 C56.33700180053711,39 45.91699981689453,30.589000701904297 45.91699981689453,30.589000701904297 C45.91699981689453,30.589000701904297 45.91699981689453,-35.01599884033203 45.91699981689453,-35.01599884033203z"
                      fill="rgb(237,246,252)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M-0.5,39 C-5.09499979019165,36.321998596191406 -13.17300033569336,32.36000061035156 -24.18199920654297,30.589000701904297 C-33.750999450683594,29.049999237060547 -41.79199981689453,29.76799964904785 -46.91699981689453,30.589000701904297 C-46.91699981689453,30.589000701904297 -46.91699981689453,-35.856998443603516 -46.91699981689453,-35.856998443603516 C-41.90999984741211,-36.79800033569336 -33.00199890136719,-37.87300109863281 -22.288000106811523,-35.856998443603516 C-12.038999557495117,-33.92900085449219 -4.635000228881836,-30.01799964904785 -0.5,-27.445999145507812 C-0.5,-27.445999145507812 -0.5,39 -0.5,39z"
                      fill="rgb(227,242,252)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M-0.5,39 C4.09499979019165,36.321998596191406 12.17300033569336,32.36000061035156 23.18199920654297,30.589000701904297 C32.750999450683594,29.049999237060547 40.79199981689453,29.76799964904785 45.91699981689453,30.589000701904297 C45.91699981689453,30.589000701904297 45.91699981689453,-35.856998443603516 45.91699981689453,-35.856998443603516 C40.90999984741211,-36.79800033569336 32.00199890136719,-37.87300109863281 21.288000106811523,-35.856998443603516 C11.038999557495117,-33.92900085449219 3.634999990463257,-30.01799964904785 -0.5,-27.445999145507812 C-0.5,-27.445999145507812 -0.5,39 -0.5,39z"
                      fill="rgb(227,242,252)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M-0.5,39 C-5.09499979019165,36.321998596191406 -13.17300033569336,32.36000061035156 -24.18199920654297,30.589000701904297 C-33.750999450683594,29.049999237060547 -41.79199981689453,29.76799964904785 -46.91699981689453,30.589000701904297 C-46.91699981689453,30.589000701904297 -46.91699981689453,-35.856998443603516 -46.91699981689453,-35.856998443603516 C-41.90999984741211,-36.79800033569336 -33.00199890136719,-37.87300109863281 -22.288000106811523,-35.856998443603516 C-12.038999557495117,-33.92900085449219 -4.635000228881836,-30.01799964904785 -0.5,-27.445999145507812 C-0.5,-27.445999145507812 -0.5,39 -0.5,39z"
                      fill="rgb(255,255,255)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M-0.5,39 C4.09499979019165,36.321998596191406 12.17300033569336,32.36000061035156 23.18199920654297,30.589000701904297 C32.750999450683594,29.049999237060547 40.79199981689453,29.76799964904785 45.91699981689453,30.589000701904297 C45.91699981689453,30.589000701904297 45.91699981689453,-35.856998443603516 45.91699981689453,-35.856998443603516 C40.90999984741211,-36.79800033569336 32.00199890136719,-37.87300109863281 21.288000106811523,-35.856998443603516 C11.038999557495117,-33.92900085449219 3.634999990463257,-30.01799964904785 -0.5,-27.445999145507812 C-0.5,-27.445999145507812 -0.5,39 -0.5,39z"
                      fill="rgb(255,255,255)"
                      fill-opacity="1"
                    />
                  </g>
                </g>
                <g
                  clip-path="url(#__lottie_element_57)"
                  style="display: none;"
                >
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                      <path />
                    </g>
                  </g>
                </g>
                <g
                  clip-path="url(#__lottie_element_38)"
                  style="display: none;"
                >
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                      <path />
                    </g>
                  </g>
                </g>
                <g
                  opacity="1"
                  style="display: block;"
                  transform="matrix(1,0,0,1,200,193.97799682617188)"
                >
                  <g
                    opacity="1"
                    transform="matrix(1,0,0,1,0,0)"
                  >
                    <path
                      d=" M-1,-28 C-1,-28 -1,39 -1,39"
                      fill-opacity="0"
                      stroke="rgb(198,226,245)"
                      stroke-linecap="butt"
                      stroke-linejoin="miter"
                      stroke-miterlimit="10"
                      stroke-opacity="1"
                      stroke-width="1"
                    />
                  </g>
                </g>
                <g
                  clip-path="url(#__lottie_element_7)"
                  style="display: none;"
                >
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  />
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                  <g
                    style="display: none;"
                  >
                    <g>
                      <path />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="max-w-xl mb-10 md:mx-auto sm:text-center lg:max-w-2xl md:mb-12"
        >
          <h2
            class="max-w-lg mb-6 font-sans text-3xl font-bold leading-none tracking-tight text-gray-900 sm:text-4xl md:mx-auto"
          >
            <span
              class="relative inline-block"
            >
              <svg
                class="absolute top-0 left-0 z-0 hidden w-32 -mt-8 -ml-20 text-blue-gray-100 lg:w-32 lg:-ml-28 lg:-mt-10 sm:block"
                fill="currentColor"
                viewBox="0 0 52 24"
              >
                <defs>
                  <pattern
                    height=".30"
                    id="e77df901-b9d7-4b9b-822e-16b2d410795b"
                    width=".135"
                    x="0"
                    y="0"
                  >
                    <circle
                      cx="1"
                      cy="1"
                      r=".7"
                    />
                  </pattern>
                </defs>
                <rect
                  fill="url(#e77df901-b9d7-4b9b-822e-16b2d410795b)"
                  height="24"
                  width="52"
                />
              </svg>
              <span
                class="relative"
              >
                Loading
              </span>
            </span>
             
            Content
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
